packages/code-studio/src/settings/FormattingSectionContent.tsx

```diff
@@ -23,45 +23,39 @@ import {
   getDefaultDateTimeFormat,
   getDefaultDecimalFormatOptions,
   getDefaultIntegerFormatOptions,
-  getFormatter,
   getTimeZone,
   getShowTimeZone,
   getShowTSeparator,
   getTruncateNumbersWithPound,
-  getSettings,
   saveSettings as saveSettingsAction,
   RootState,
   WorkspaceSettings,
 } from '@deephaven/redux';
+import { assertNotNull } from '@deephaven/utils';
 import './FormattingSectionContent.scss';
 import type { DebouncedFunc } from 'lodash';
 import {
   focusFirstInputInContainer,
   isSameDecimalOptions,
   isSameIntegerOptions,
   isValidFormat,
-  removeFormatRuleExtraProps,
-  isFormatRuleValidForSave,
-  ValidFormatterItem,
 } from './SettingsUtils';
-import type { FormatterItem, FormatOption } from './SettingsUtils';
+import type { FormatOption } from './SettingsUtils';
 import DateTimeOptions from './DateTimeOptions';
 import TimeZoneOptions from './TimeZoneOptions';
 
 const log = Log.module('FormattingSectionContent');
 
 interface FormattingSectionContentProps {
   dh: DhType;
-  formatter: FormatterItem[];
-  defaultDateTimeFormat: string;
-  showTimeZone: boolean;
-  showTSeparator: boolean;
-  timeZone: string;
-  truncateNumbersWithPound: boolean;
-  settings: WorkspaceSettings;
-  saveSettings: (settings: WorkspaceSettings) => void;
-  defaultDecimalFormatOptions: FormatOption;
-  defaultIntegerFormatOptions: FormatOption;
+  defaultDateTimeFormat?: string;
+  showTimeZone?: boolean;
+  showTSeparator?: boolean;
+  timeZone?: string;
+  truncateNumbersWithPound?: boolean;
+  saveSettings: (settings: Partial<WorkspaceSettings>) => void;
+  defaultDecimalFormatOptions?: FormatOption;
+  defaultIntegerFormatOptions?: FormatOption;
```

Comment:
This was mostly wrestling with TypeScript. I had to change the Redux state type for `workspace.data.settings` to be optional (because it really is optional now), so the getWorkspace's return value is now typed as CustomizableWorkspace instead of Workspace, marking these parameters optional. I'll investigate for a workaround